akka-http/src/main/scala/akka/http/scaladsl/server/directives/ParameterDirectives.scala

```diff
@@ -79,12 +87,29 @@ object ParameterDirectives extends ParameterDirectives {
   private val _parameterSeq: Directive1[immutable.Seq[(String, String)]] =
     extract(_.request.uri.query().toSeq)
 
+  trait ParamSpec {
+    type Out
+    def apply(): Directive1[Out]
```

Comment:
This is the important difference to make it work: we require here that each parameter argument must provide exactly one value. This almost works out of the box, only for required parameters there's a slight syntax change because they used to result in a `Directive0` where they are now a somewhat weird `Directive1[Unit]`.